component/component/common.libsonnet

```
@@ -173,6 +173,97 @@ local argoCDAnnotations() = {
   'argocd.argoproj.io/sync-options': 'Prune=false',
 };
 
+local bottomPod(query) = 'label_replace( bottomk(1, %(query)s) * on(namespace) group_left(label_appcat_vshn_io_claim_namespace) kube_namespace_labels, "name", "$1", "namespace", "vshn-replacemeplease-(.+)-.+")' % query;
+local topPod(query) = 'label_replace( topk(1, %(query)s) * on(namespace) group_left(label_appcat_vshn_io_claim_namespace) kube_namespace_labels, "name", "$1", "namespace", "vshn-replacemeplease-(.+)-.+")' % query;
+
+local generatePrometheusNonSLORules(serviceName, memoryContainerName, additionalAlertsRuleGroup) = {
+  // standardized lowercase regardless of what came as input
+  local serviceNameLower = std.asciiLower(serviceName),
+  local toReplace = 'vshn-replacemeplease',
+  local queries = {
+    availableStorage: 'kubelet_volume_stats_available_bytes{job="kubelet", metrics_path="/metrics"}',
+    availablePercent: '(%s / kubelet_volume_stats_capacity_bytes{job="kubelet", metrics_path="/metrics"})' % queries.availableStorage,
+    usedStorage: 'kubelet_volume_stats_used_bytes{job="kubelet", metrics_path="/metrics"}',
+    unlessExcluded: 'unless on(namespace, persistentvolumeclaim) kube_persistentvolumeclaim_access_mode{ access_mode="ReadOnlyMany"} == 1 unless on(namespace, persistentvolumeclaim) kube_persistentvolumeclaim_labels{label_excluded_from_alerts="true"} == 1',
+  },
+  name: 'prometheusrule',
+  base: {
+
+    apiVersion: 'kubernetes.crossplane.io/v1alpha1',
+    kind: 'Object',
+    metadata: {
+      name: 'prometheusrule',
+    },
+    spec: {
+      providerConfigRef: {
+        name: 'kubernetes',
+      },
+      forProvider+: {
+        manifest+: {
+          metadata: {
+            apiVersion: 'monitoring.coreos.com/v1',
+            kind: 'PrometheusRule',
```

```suggestion
          apiVersion: 'monitoring.coreos.com/v1',
          kind: 'PrometheusRule',
          metadata: {
```

Afaik `apiVersion` and `kind` need to be on the same level as `metadata` and not inside of it.